x-pack/plugins/secops/server/lib/hosts/elasticsearch_adapter.ts

```diff
@@ -21,21 +21,28 @@ export class ElasticsearchHostsAdapter implements HostsAdapter {
       'search',
       buildQuery(options)
     );
-    const { limit } = options.pagination;
+    const { cursor, limit } = options.pagination;
     const totalCount = getOr(0, 'aggregations.host_count.value', response);
     const hits: HostHit[] = getOr([], 'aggregations.group_by_host.buckets', response).map(
-      (bucket: HostBucket) => ({ ...head(bucket.host.hits.hits), cursor: bucket.key!.host_name })
+      (bucket: HostBucket) => ({
+        ...head(bucket.host.hits.hits),
+        cursor: bucket.key!.host_name,
+        firstSeen: bucket.firstSeen!.value_as_string,
+      })
     );
     const hostsEdges = hits.map(hit => formatHostsData(options.fields, hit, hostsFieldsMap));
     const hasNextPage = hostsEdges.length === limit + 1;
-    const edges = hasNextPage ? hostsEdges.splice(0, limit) : hostsEdges;
-    const lastCursor = get('cursor', last(edges));
+    const beginning = cursor != null ? parseInt(cursor!, 10) : 0;
```

Comment:
Since you're checking if `cursor != null`, Typescript 👼does not require you to list the `!`  ❗️ . You can remove the bang now.

Reply:
Perfect! I went ahead and removed the extra `!`. This was a copypasta to fix pagination coming from `UncommonProcesses`'s [elasticsearch_adapter.ts](https://github.com/elastic/kibana/blob/feature-secops/x-pack/plugins/secops/server/lib/uncommon_processes/elasticsearch_adapter.ts#L43).